src/web/assets/cp/dist/js/Craft.js

```diff
@@ -12652,8 +12677,47 @@ Craft.EditableTable = Garnish.Base.extend(
                 this.addListener(Garnish.$win, 'resize', 'initializeIfVisible');
             }
         },
+        updateAddRowButton: function() {
+            if ((this.hasMaxRows && this.rowCount >= this.maxRows) ||
+                (this.hasMinRows && this.rowCount < this.minRows)) {
+                this.$addRowBtn.css('opacity', '0.2');
+            } else {
+                this.$addRowBtn.css('opacity', '1');
+            }
+        },
+        canDeleteRow: function() {
+            return (this.rowCount > this.minRows);
+        },
+        deleteRow: function(row) {
+            if (this.hasMaxRows && this.hasMinRows) {
+                if (!this.canDeleteRow()) {
+                    return;
+                }
+            }
+
+            this.sorter.removeItems(row.$tr);
+            row.$tr.remove();
+
+            if (this.hasMinRows && this.hasMinRows) {
+                this.rowCount--;
+            }
 
+            this.updateAddRowButton();
+            // onDeleteRow callback
+            this.settings.onDeleteRow(row.$tr);
+        },
+        canAddRow: function() {
+            return (this.rowCount < this.maxRows);
+        },
         addRow: function() {
+            if (this.hasMaxRows && this.hasMinRows) {
```

Comment:
@Radabaugh not sure if you need to check hasMinRows when you are only looking to add a row at this point